table/table_builder.cc

```diff
@@ -199,7 +202,30 @@ void TableBuilder::WriteBlock(BlockBuilder* block, BlockHandle* handle) {
         r->sst_counters.Inc(eSstCountCompressAborted);
       }
       break;
-    }
+
+    case kLZ4Compression:
```

Comment:
It might be a larger refactoring than you're willing to make in the context of this PR, but it seems like this case statement and others that look very much like it could potentially be replaced with polymorphism to good effect.